### ISSUE TITLE
How to not allow PR's merged in Staging to get merged in Prod!

### ISSUE THREAD

**Issue by balajialg** [berkeley-dsep-infra/hubs#2889]
If we want to play around with staging environment without making changes to prod/ensuring that the changes get merged in prod, what would be an effective way forward?

**Comment by yuvipanda** [comment]
I think this depends on the kinds of changes. What kinds of changes do you have in mind?

**Comment by felder** [comment]
@yuvipanda the idea would be to permit me to make changes to things to grafana or perhaps the hubs that would require testing over multiple days without holding up other deploys to prod.  Essentially is there a way to cherry pick commits to staging for deployments to prod?

**Comment by yuvipanda** [comment]
@felder the way I do this is to deploy from my local machine to particular staging instances. It'll get wiped everytime anyone merges a PR though, but then I just deploy again and go as it just takes a minute or two. What do you think of that?

**Comment by felder** [comment]
@yuvipanda that will probably work.  I may need a short tutorial on hubploy to make sure I know how to do this.

**Comment by yuvipanda** [comment]
@felder makes sense!

I think it would be:

1. Make sure pre-reqs are installed (`pip install -r requirements.txt`)
2. Make sure you have `gcloud` installed, and are authenticated correctly so `sops` works
3. Run `hubploy deploy <hub-name> hub <staging|prod>` to do a deploy
4. Make changes, repeat (3) as needed

Can you try that out?

**Comment by ryanlovett** [comment]
@yuvipanda That worked for me back in the git-crypt days, but not with sops. hubploy wants things to be decrypted. CI decrypts in place at various stages with `sops -d -i support/secrets.yaml`.

Since hubploy ultimately runs helm with the path to deployments/{deployment}/secrets/staging.yaml, I suppose you can manually replace that with process substitution, e.g.  `helm ... -f <(sops -d deployments/{deployment}/secrets/staging.yaml)`

**Comment by yuvipanda** [comment]
@ryanlovett are you on the latest version of hubploy? I added sops support to hubploy so you wouldn't have to do that.

**Comment by ryanlovett** [comment]
@yuvipanda No, sorry, I should have checked!

**Comment by ryanlovett** [comment]
Per @yuvipanda 's advice in slack, one should install hubploy from github, https://github.com/yuvipanda/hubploy.

**Comment by felder** [comment]
I tried to use hubploy to deploy biology-staging yesterday, I was not successful but also have not yet tried other hubs or dug into why it failed.

**Comment by yuvipanda** [comment]
@felder cool, let me know what errors you run into and I'll see how I can help! You'll need to install it from git, not from pypi

**Comment by felder** [comment]
@yuvipanda yeah did the git installation and before roping you in I definitely wanted to confirm myself whether or not it was chair to computer interface problem.
